processing/src/main/java/io/druid/segment/ColumnSelectorFactory.java

```diff
@@ -27,6 +27,7 @@
 public interface ColumnSelectorFactory
 {
   public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec);
+  public DimensionSelector makeDictEncodedStringDimensionSelector(DimensionSpec dimensionSpec);
```

Comment:
I don't understand why this method is being introduced.  It returns the same thing as `makeDimensionSelector`.  If there is a semantic difference to what is being returned, it's likely time for some javadoc.  Also, why not introduce a new interface?